core/src/main/java/org/apache/iceberg/io/BaseTaskWriter.java

```diff
@@ -75,6 +79,113 @@ public WriteResult complete() throws IOException {
         .build();
   }
 
+  /**
+   * Base delta writer to write both insert records and equality-deletes.
+   */
+  protected abstract class BaseDeltaWriter implements Closeable {
```

Comment:
How about `BaseEqualityDeltaWriter`? I think Spark `MERGE INTO` will likely use a delta writer that doesn't create the equality writer or use the `SortedPosDeleteWriter` because it will request that the rows are already ordered.